packages/perseus-score/src/score.test.ts

```diff
@@ -0,0 +1,613 @@
+            "expression 1": "2+2",
+        };
+
+        // Act
+        const result = scoreWidgetsFunctional(
+            widgets,
+            widgetIds,
+            userInputMap,
+
+            "en",
+        );
+
+        // Assert
+        expect(result["expression 1"]).toHaveBeenAnsweredCorrectly();
+    });
+
+    it("can handle a correct legacy Expression widget", () => {
+        // Arrange
+        const widgets: PerseusWidgetsMap = {
+            "expression 1": getLegacyExpressionWidget() as any,
+        };
+        const widgetIds: Array<string> = ["expression 1"];
+        const userInputMap: UserInputMap = {
+            "expression 1": "2+2",
+        };
+
+        // Act
+        const result = scoreWidgetsFunctional(
+            widgets,
+            widgetIds,
+            userInputMap,
+
+            "en",
+        );
+
+        // Assert
+        expect(result["expression 1"]).toHaveBeenAnsweredCorrectly();
+    });
+
+    it("can handle an incorrect modern Expression widget", () => {
+        // Arrange
+        const widgets: PerseusWidgetsMap = {
+            "expression 1": getExpressionWidget(),
+        };
+        const widgetIds: Array<string> = ["expression 1"];
+        const userInputMap: UserInputMap = {
+            "expression 1": "2+42",
+        };
+
+        // Act
+        const result = scoreWidgetsFunctional(
+            widgets,
+            widgetIds,
+            userInputMap,
+
+            "en",
+        );
+
+        // Assert
+        expect(result["expression 1"]).toHaveBeenAnsweredIncorrectly();
+    });
+
+    it("can handle an incorrect legacy Expression widget", () => {
+        // Arrange
+        const widgets: PerseusWidgetsMap = {
+            "expression 1": getLegacyExpressionWidget() as any,
+        };
+        const widgetIds: Array<string> = ["expression 1"];
+        const userInputMap: UserInputMap = {
+            "expression 1": "2+42",
+        };
+
+        // Act
+        const result = scoreWidgetsFunctional(
+            widgets,
+            widgetIds,
+            userInputMap,
+
+            "en",
+        );
+
+        // Assert
+        expect(result["expression 1"]).toHaveBeenAnsweredIncorrectly();
+    });
+});
+
+function generateDropdown(): DropdownWidget {
+    return {
+        type: "dropdown",
+        alignment: "default",
+        static: false,
+        graded: true,
+        options: {
+            static: false,
+            ariaLabel: "Test ARIA label",
+            visibleLabel: "Test visible label",
+            placeholder: "Answer me",
+            choices: [
+                {
+                    content: "Incorrect",
+                    correct: false,
+                },
+                {
+                    content: "Correct",
+                    correct: true,
+                },
+            ],
+        },
+        version: {
+            major: 0,
+            minor: 0,
+        },
+    };
+}
+
+describe("scorePerseusItem", () => {
```

Comment:
Jeremy and I refactored a these tests to not use mocks and instead rely on real data.